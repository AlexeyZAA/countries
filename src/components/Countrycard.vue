<template>
    <v-card class="mx-auto" max-width="250px">
      <v-img
        :src="flag"
        height="150px"
        width="250px"
      ></v-img>
      <v-card-actions>
        <v-spacer>
          <div class="titlbold">
            {{ name }}
            <router-link to="/about">О нас</router-link>
            </div>
        </v-spacer>
        <v-btn icon @click="showModal">
          <v-icon>mdi-pencil-box-outline</v-icon>
        </v-btn>
      </v-card-actions>

      <v-card-text class="text--primary">
        <v-simple-table dense>
          <tbody>
            <tr>
              <td class="titlbold">Столица</td>
              <td>{{ capital }}</td>
            </tr>
            <tr>
              <td class="titlbold">Регион</td>
              <td>{{ region }}</td>
            </tr>
            <tr>
              <td class="titlbold">Население</td>
              <td>{{ population }}</td>
            </tr>
          </tbody>
        </v-simple-table>
      </v-card-text>
    </v-card>
</template>

<script>
export default {
  name: 'Countrycard',
  props: ['flag', 'name', 'capital', 'region', 'population', 'hide'],
  data() {
    return {
    }
  },
  methods: {
    showModal() {
       this.$store.dispatch('SETMODALDATA', {
        name: this.name, 
        capital: this.capital, 
        region: this.region,
        population: this.population
        })
      this.$store.dispatch('SETDIALOG', true)
    },
    saveCountryItem() {
      this.$store.dispatch('UPDATECOUNTRY', this.formUpdate)
      this.formUpdate = {
        name: '',
        capital: '',
        region: '',
        population: '',
      }
      this.dialog = false
    },
  },
}
</script>

<style scoped>
.headLeft {
  display: inline-block;
  left: 10px;
  position: absolute;
}
.titlbold {
  font-weight: bold;
}
</style>
