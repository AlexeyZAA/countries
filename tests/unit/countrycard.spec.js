import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import Countrycard from '@/components/Countrycard.vue'

describe('Блок тестов Countrycard.vue', () => {
  it('Снимок краты страны', () => {
    const flag = 'http://testdomain'
    const name = 'Текст имя'
    const  capital = 'Текст столица'
    const  region = 'Текст регион'
    const  population = 'Текст население'
    const renderer = createRenderer()
    const hide = false

    const wrapper = shallowMount(Countrycard, {
    propsData: {
        flag, name, capital, region, population, hide 
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})