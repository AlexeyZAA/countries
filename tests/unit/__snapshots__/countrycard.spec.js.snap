// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Блок тестов Countrycard.vue Снимок краты страны 1`] = `
<v-card max-width="250px" class="mx-auto">
  <v-img src="http://testdomain" height="150px" width="250px"></v-img>
  <v-card-actions>
    <v-spacer>
      <div class="titlbold">Текст имя</div>
    </v-spacer>
    <v-btn icon="">
      <v-icon>mdi-pencil-box-outline</v-icon>
    </v-btn>
  </v-card-actions>
  <v-card-text class="text--primary">
    <v-simple-table dense="">
      <tbody>
        <tr>
          <td class="titlbold">Столица</td>
          <td>Текст столица</td>
        </tr>
        <tr>
          <td class="titlbold">Регион</td>
          <td>Текст регион</td>
        </tr>
        <tr>
          <td class="titlbold">Население</td>
          <td>Текст население</td>
        </tr>
      </tbody>
    </v-simple-table>
  </v-card-text>
</v-card>
`;
